crates/rpc/rpc-server-types/src/result.rs

```diff
@@ -148,6 +150,24 @@ pub fn rpc_err(
     )
 }
 
+/// Formats a [`BlockId`] into an error message.
+// todo: impl Display for `alloy_eips::eip1898::BlockId`
+pub fn block_id_to_str(id: BlockId) -> String {
+    match id {
+        BlockId::Hash(h) => {
+            if h.require_canonical == Some(true) {
+                format!("canonical hash {}", h.block_hash)
+            } else {
+                format!("hash {}", h.block_hash)
+            }
+        }
+        BlockId::Number(n) => match n {
+            BlockNumberOrTag::Number(n) => format!("number {}", n),
+            _ => format!("{}", n),
+        },
+    }
+}
+
 #[cfg(test)]
 mod tests {
```

Comment:
can we add some sanity tests here that format the notfound errors?

I'd like to have full error strings in the codebase so this can be found with ease